src/core/search/ast_expr.cc

```diff
@@ -42,4 +44,16 @@ AstFieldNode::AstFieldNode(string field, AstNode&& node)
     : field{field.substr(1)}, node{make_unique<AstNode>(move(node))} {
 }
 
+AstTagsNode::AstTagsNode(std::string tag) {
+  tags = {move(tag)};
+}
+
+AstTagsNode::AstTagsNode(AstExpr&& l, std::string tag) {
```

Comment:
You implemented tag addition in a very peculiar way. Is it because how you handle variant?

Reply:
No, only for simplicity in the parser declaration. Its cumbersome to manage state there, so I do it here